kore/src/Kore/Repl/Data.hs

```diff
@@ -494,14 +495,23 @@ deriving instance MonadSMT m => MonadSMT (UnifierWithExplanation m)
 
 deriving instance MonadProfiler m => MonadProfiler (UnifierWithExplanation m)
 
+instance MonadTrans UnifierWithExplanation where
+    lift = UnifierWithExplanation . lift . lift
+
 instance MonadLog m => MonadLog (UnifierWithExplanation m) where
     logEntry entry = UnifierWithExplanation $ logEntry entry
     {-# INLINE logEntry #-}
     logWhile entry ma =
         UnifierWithExplanation $ logWhile entry (getUnifierWithExplanation ma)
     {-# INLINE logWhile #-}
 
-deriving instance MonadSimplify m => MonadSimplify (UnifierWithExplanation m)
+instance MonadSimplify m => MonadSimplify (UnifierWithExplanation m) where
```

Comment:
Why can't we use the default implementations anymore?

Reply:
When trying to derive the implementation, the compiler error is:
```    
        Couldn't match type ‘UnifierWithExplanation m’
                     with ‘UnifierT (AccumT (First ReplOutput) m)’
        arising from a use of ‘ghc-prim-0.5.3:GHC.Prim.coerce’
```
Do you know how this could be fixed?